File: spark-in-action/java/src/main/java/ingestion/other/PhotoMetadataIngestionApp.java
package ingestion.other;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class PhotoMetadataIngestionApp {
    public static void main(String[] args) {
        PhotoMetadataIngestionApp app = new PhotoMetadataIngestionApp();
        app.start();
    }

    private boolean start() {
        SparkSession spark = SparkSession.builder()
                .appName("EXIF to Dataset")
                .master("local")
                .getOrCreate();

        spark.sparkContext().setLogLevel("ERROR");

        String importDirectory = "data/ch09";

        Dataset<Row> df = spark.read()
                .format("exif")
                .option("recursive", "true")  // be able to read recursively through the directories
                .option("limit", "100000")  // limit files
                .option("extensions", "jpg, jpeg")
                .load(importDirectory);

        System.out.println("Import " + df.count() + " photos");
        df.printSchema();
        df.show(5);

        return true;
    }
}

File: spark-in-action/scala/src/main/scala/ingestion/other/PhotoMetadataIngestionApp.scala
package ingestion.other

import org.apache.spark.sql.SparkSession


/**
 * Ingest metadata from a directory containing photos, make them available as EXIF.
 *
 */
object PhotoMetadataIngestionApp {
  def main(args: Array[String]): Unit = {
    val spark = SparkSession.builder
      .appName("EXIF to Dataset")
      .master("local")
      .getOrCreate

    val importDirectory = "data/ch09"

    val df = spark.read
      .format("exif")
      .option("recursive", "true")
      .option("limit", "100000")
      .option("extensions", "jpg,jpeg")
      .load(importDirectory)

    println("imported " + df.count + " photos.")
    df.printSchema()
    df.show(5)
  }
}
